src/android/patches/0.20/makeMainActivityPatch.js

```diff
@@ -3,10 +3,13 @@ const addPackagePatch = require('./addPackagePatch');
 const append = (scope, pattern, patch) =>
   scope.replace(pattern, `${pattern}${patch}`);
 
-module.exports = function makeMainActivityPatch(config) {
+module.exports = function makeMainActivityPatch(androidConfig, params) {
   const importPattern = 'import com.facebook.react.ReactActivity;';
   const packagePattern = 'new MainReactPackage()';
 
+  const packageInstance = androidConfig.packageInstance
+    .replace(/\$\{(\w+)\}/g, (pattern, paramName) => `"${params[paramName]}"`);
```

Comment:
I don't think we need to allocate a standalone file just for regexp. Besides that, even if we'll do it, there are some differences (as you said), like nil vs null. 

We can test the whole functionality in one place without creating a tons of 1-10-line files, so I don't see a point there also.